include/alpaka/warp/Traits.hpp

```diff
@@ -0,0 +1,193 @@
+        //! Evaluates predicate for all active threads of the warp and returns
+        //! non-zero if and only if predicate evaluates to non-zero for any of them.
+        //!
+        //! It follows the logic of __any(predicate) in CUDA before version 9.0 and HIP,
+        //! the operation is applied for all active threads.
+        //! The modern CUDA counterpart would be __any_sync(__activemask(), predicate).
+        //!
+        //! \tparam TWarp The warp implementation type.
+        //! \param warp The warp implementation.
+        //! \param predicate The predicate value for current thread.
+        ALPAKA_NO_HOST_ACC_WARNING
+        template<
+            typename TWarp>
+        ALPAKA_FN_ACC auto any(
+            TWarp const & warp,
+            std::int32_t predicate)
+        -> std::int32_t
+        {
+            using ImplementationBase = concepts::ImplementationBase<ConceptWarp, TWarp>;
+            return traits::Any<
+                ImplementationBase>
+            ::any(
+                warp,
+                predicate);
+        }
+
+        //-----------------------------------------------------------------------------
+        //! Evaluates predicate for all non-exited threads in mask and returns
+        //! a 64-bit integer whose Nth bit is set if and only if predicate
+        //! evaluates to non-zero for the Nth thread of the warp and the Nth
+        //! thread is active.
+        //!
+        //! It follows the logic of __ballot(predicate) in CUDA before version 9.0 and HIP,
+        //! the operation is applied for all active threads.
+        //! The modern CUDA counterpart would be __ballot_sync(__activemask(), predicate).
+        //! Return type is 64-bit to fit all platforms.
+        //!
+        //! \tparam TWarp The warp implementation type.
+        //! \param warp The warp implementation.
+        //! \param predicate The predicate value for current thread.
+        ALPAKA_NO_HOST_ACC_WARNING
+        template<
+            typename TWarp>
+        ALPAKA_FN_ACC auto ballot(
+            TWarp const & warp,
+            std::int32_t predicate)
+        -> std::uint64_t
+        {
+            using ImplementationBase = concepts::ImplementationBase<ConceptWarp, TWarp>;
+            return traits::Ballot<
+                ImplementationBase>
+            ::ballot(
+                warp,
+                predicate);
+        }
+
+        //-----------------------------------------------------------------------------
+        //! Returns a 64-bit integer whose Nth bit is set if and only if the Nth
+        //! thread of the warp is active.
+        //!
+        //! \tparam TWarp The warp implementation type.
+        //! \param warp The warp implementation.
+        ALPAKA_NO_HOST_ACC_WARNING
+        template<
+            typename TWarp>
+        ALPAKA_FN_ACC auto activemask(
+            TWarp const & warp)
+        -> std::uint64_t
```

Comment:
we should not enforce a 64bit activemask. The retuen value should be implementaion dependant. 32bit for CUDA and 64bit for HIP
The comment should have a doxygen \return 64bit or 32bit type depending on the accelerator.

Reply:
I disagree. I think it's much easier for a user to work with a fixed type, and not have conditional types depending on the backend.

Reply:
And with your suggestion, a user is forced to write those ugly `static_cast`s that we have inside already.

Reply:
A user can use `auto mask = alpaka::activeMask()`. The point is we waist here register and performance.
A typically use case for active mask is `uint32_t threads = alpaka::popc(mask)`. 
Do you have a use case in mind where a static cast is required?

Reply:
Discussed offline, I will implement this suggestion.